query/src/test/java/org/infinispan/query/distributed/MassIndexingTest.java

```diff
@@ -56,51 +60,58 @@ public void testReindexing() throws Exception {
 
    public void testOverlappingMassIndexers() {
       Cache<Integer, Car> cache = cache(0);
-      SearchManager searchManager = Search.getSearchManager(cache);
-      MassIndexer massIndexer = searchManager.getMassIndexer();
-
       IntStream.range(0, 10).forEach(i -> cache.put(i, new Car("whatever", "whatever", 0)));
 
+      Indexer massIndexer = Search.getIndexer(cache);
+
+      CountDownLatch latch = new CountDownLatch(1);
+      instrumentIndexer(massIndexer, latch);
 
-      CompletionStage<Void> first = massIndexer.startAsync();
-      eventually(() -> {
-         log.debug("Checking if massIndexer is running");
-         return massIndexer.isRunning();
-      }, 10_000, 10, TimeUnit.MILLISECONDS);
+      CompletionStage<Void> first = massIndexer.run();
+      CompletionStage<Void> second = massIndexer.run();
 
-      CompletionStage<Void> second = massIndexer.startAsync();
+      latch.countDown();
 
-      assertSuccessCompletion(first);
-      assertErrorCompletion(second, MassIndexerAlreadyStartedException.class);
-      eventually(() -> !massIndexer.isRunning());
+      assertTrue((isSuccess(first) && isError(second)) || (isError(second) && isSuccess(first)));
```

Comment:
This condition is quite surely wrong, because the right side is identical to the left side of the OR

Reply:
yes, it's wrong, but I can see you fixed it